pkgs/development/libraries/openmpi/default.nix

```diff
@@ -1,136 +1,247 @@
+    HOSTNAME = "localhost";
+    SOURCE_DATE_EPOCH = "0";
+  };
 
-  buildInputs = [ zlib ]
-    ++ lib.optionals stdenv.isLinux [ libnl numactl pmix ucx ucc ]
+  outputs = [
+    "out"
+  ] ++ lib.optionals stdenv.isLinux [
+    "man"
+    "dev"
+  ];
+
+  buildInputs =
+    [
+      zlib
+      libevent
+      hwloc
+    ]
+    ++ lib.optionals stdenv.isLinux [
+      libnl
+      numactl
+      pmix
+      ucx
+      ucc
+      prrte
+    ]
     ++ lib.optionals cudaSupport [ cudaPackages.cuda_cudart ]
-    ++ [ libevent hwloc ]
-    ++ lib.optional (stdenv.isLinux || stdenv.isFreeBSD) rdma-core
-    ++ lib.optionals fabricSupport [ libpsm2 libfabric ];
-
-  nativeBuildInputs = [ perl removeReferencesTo makeWrapper ]
+    ++ lib.optionals (stdenv.isLinux || stdenv.isFreeBSD) [ rdma-core ]
+    # needed for internal pmix
+    ++ lib.optionals (!stdenv.isLinux) [ python3 ]
+    ++ lib.optionals fabricSupport [
+      libpsm2
+      libfabric
+    ];
+
+  nativeBuildInputs =
+    [
+      perl
+      removeReferencesTo
+      makeWrapper
+    ]
     ++ lib.optionals cudaSupport [ cudaPackages.cuda_nvcc ]
     ++ lib.optionals fortranSupport [ gfortran ];
 
-  configureFlags = lib.optional (!cudaSupport) "--disable-mca-dso"
-    ++ lib.optional (!fortranSupport) "--disable-mpi-fortran"
-    ++ lib.optionals stdenv.isLinux  [
-      "--with-libnl=${lib.getDev libnl}"
-      "--with-pmix=${lib.getDev pmix}"
-      "--with-pmix-libdir=${pmix}/lib"
-      "--enable-mpi-cxx"
-    ] ++ lib.optional enableSGE "--with-sge"
-    ++ lib.optional enablePrefix "--enable-mpirun-prefix-by-default"
+  configureFlags = [
+    (lib.enableFeature cudaSupport "mca-dso")
+    (lib.enableFeature fortranSupport "mpi-fortran")
+    (lib.withFeatureAs stdenv.isLinux "libnl" (lib.getDev libnl))
+    "--with-pmix=${if stdenv.isLinux then (lib.getDev pmix) else "internal"}"
+    (lib.withFeatureAs stdenv.isLinux "pmix-libdir" "${lib.getLib pmix}/lib")
+    # Puts a "default OMPI_PRTERUN" value to mpirun / mpiexec executables
+    (lib.withFeatureAs stdenv.isLinux "prrte" (lib.getBin prrte))
+    (lib.withFeature enableSGE "sge")
+    (lib.enableFeature enablePrefix "mpirun-prefix-by-default")
     # TODO: add UCX support, which is recommended to use with cuda for the most robust OpenMPI build
     # https://github.com/openucx/ucx
     # https://www.open-mpi.org/faq/?category=buildcuda
-    ++ lib.optionals cudaSupport [ "--with-cuda=${lib.getDev cudaPackages.cuda_cudart}" "--enable-dlopen" ]
-    ++ lib.optionals fabricSupport [ "--with-psm2=${lib.getDev libpsm2}" "--with-libfabric=${lib.getDev libfabric}" ]
-    ;
+    (lib.withFeatureAs cudaSupport "cuda" (lib.getDev cudaPackages.cuda_cudart))
+    (lib.enableFeature cudaSupport "dlopen")
+    (lib.withFeatureAs fabricSupport "psm2" (lib.getDev libpsm2))
+    (lib.withFeatureAs fabricSupport "ofi" (lib.getDev libfabric))
+    (lib.withFeatureAs fabricSupport "ofi-libdir" "${lib.getLib libfabric}/lib")
```

Comment:
> configure: error: "yes" or "no" are not valid arguments for --with-ofi-libdir

on aarch64-linux this seems to yield "yes" or "no" !?

Reply:
> > configure: error: "yes" or "no" are not valid arguments for --with-ofi-libdir
> 
> on aarch64-linux this seems to yield "yes" or "no" !?

No, but the configure script parses `--without-ofi-libdir` in an equivalent way to `--with-ofi-libdir=no` which apparently is not supported.. I switched to using `lib.optionals` only for this option.